pkg/build/builder/util.go

```diff
@@ -26,3 +30,67 @@ func getBuildEnvVars(build *api.Build) map[string]string {
 	}
 	return envVars
 }
+
+type BuildConfigClientInterface interface {
+	GetBuildConfig(namespace, name string) (*buildapi.BuildConfig, error)
+	UpdateBuildConfig(buildConfig *buildapi.BuildConfig) error
+}
+
+type BuildClientInterface interface {
```

Comment:
CreateBuildFromConfig and CreateBuildWithImmutableId need to be interfaces somewhere so they can be mocked...  I might be inclined to agree that the methods that "just" create the object w/ no other logic should be separated from the more sophisticated paths, though.